ood-portal-generator/templates/ood-portal.conf.erb

```diff
@@ -267,12 +267,23 @@ Listen <%= addr_port %>
     <%- end -%>
   </Location>
 
-  <%- if @api_cors_enabled -%>
-  # CORS setup for API requests
-  <Location "<%= @api_cors_uri %>">
+  <%- if @api_auth_enabled || @api_cors_enabled -%>
+  # Setup for API requests
+  <Location "<%= @api_uri %>">
+    RewriteEngine On
+    <%- if @api_auth_enabled -%>
+    # Client API auth check
+    RewriteCond "%{HTTP:X-OOD-API-ID}" !^[a-z0-9-]+$ [NC,OR]
+    RewriteCond "%{HTTP:X-OOD-API-CLIENT-SECRET}" !^[a-z0-9-]+$ [NC,OR]
+    RewriteCond "<%= @api_auth_root %>/clients/%{HTTP:X-OOD-API-ID}.%{HTTP:X-OOD-API-CLIENT-SECRET}" !-f
+    RewriteRule ^.*$ - [R=401,L]
```

Comment:
I think this should be `403` because you're authenticated, just not authorized.

Reply:
I do not mind either way, but we have selected 401 as the credentials will not be provided or invalid when the condition is met.

My understanding is that 403 is when your credentials are valid, but they do not give you access to the resource you are trying to access.

What shall I use?

Reply:
I see, yea these are credentials of a sort, but I guess I'm looking for way to distinguish this error and needing to refresh credentials (like OIDC token needing to refresh and so on).

Maybe 403 for now and we can decide on something better later?  You're writing a client right? Wouldn't you need some way to distinguish between needing to re-authenticate and this (needing to refresh your client id and/or secret?)

Reply:
Yes, that is a good point, it would be much easier returning 403 to distinguish between the 2 errors.

I have updated.